src/compact.rs

```diff
@@ -273,7 +285,12 @@ where
 
 impl<T: 'static> HasCompact for T
 where
-	Compact<T>: for<'a> EncodeAsRef<'a, T> + Decode + From<Self> + Into<Self> + MaybeMaxEncodedLen,
+	Compact<T>: for<'a> EncodeAsRef<'a, T>
+		+ Decode
+		+ DecodeWithMemTracking
```

Comment:
This and the change above are semver breaking and require a major bump.

Reply:
Hmmm yes, you're right, good catch ! But, I wouldn't be very keen on bumping the major version, since then we would also have to implement https://github.com/paritytech/parity-scale-codec/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22SCALE+codec+4.0%22 in order to do the release, and it would delay this feature. So WDYT about https://github.com/paritytech/parity-scale-codec/pull/616/commits/336c813bec1c27f4b413a536824ad46dc1be0865 as an alternative ? At least temporary until the other work for releasing 4.0 is done ?

Reply:
@bkchr sorry for the ping. Can you take a look on the answer above please ?